app/lib/schema-data.server.ts

```diff
@@ -118,53 +118,59 @@ export type GitContentDataResponse = {
   children?: GitContentDataResponse[]
 }
 
-export async function loadSchemaExamples(
-  path: string,
-  ref: string
-): Promise<ExampleFiles[]> {
-  const dirPath = path.substring(0, path.lastIndexOf('/'))
-  const schemaName = path.substring(path.lastIndexOf('/') + 1)
-  const exampleFiles = (await getGithubDir(dirPath, ref)).filter(
-    (x) =>
-      x.name.startsWith(`${schemaName.split('.')[0]}.`) &&
-      x.name.endsWith('.example.json')
-  )
-
-  const result: ExampleFiles[] = []
-  for (const exampleFile of exampleFiles) {
-    const exPath = join(dirPath, '/', exampleFile.name)
-    const example = await loadContentFromGithub(exPath)
-    result.push({
-      name: exampleFile.name.replace('.example.json', ''),
-      content: example,
-    })
-  }
-  return result
-}
-
-export async function getGithubDir(
-  path?: string,
-  ref = 'main'
-): Promise<GitContentDataResponse[]> {
-  return (
-    await octokit.repos.getContent({
-      ...repoData,
-      path: path ?? 'gcn',
-      ref,
-    })
-  ).data as GitContentDataResponse[]
-}
+export const loadSchemaExamples = memoizee(
+  async function (path: string, ref: string): Promise<ExampleFiles[]> {
+    const dirPath = path.substring(0, path.lastIndexOf('/'))
+    const schemaName = path.substring(path.lastIndexOf('/') + 1)
```

Comment:
Use `path.dirname` and `path.basename` rather than manual string processing. https://nodejs.org/api/path.html